src/core/rateLimiter.js

```
@@ -15,131 +15,139 @@ class RateLimiter {
+
         this.requestCounts = {
             github: { count: 0, resetTime: null },
             gitlab: { count: 0, resetTime: null }
         };
+
+        this.countdownInterval = null;
     }
 
     // Set API token - updateHeader logic
     setToken(token, platform) {
+        if (!this.tokens.hasOwnProperty(platform)) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)